src/aiida_quantumespresso/workflows/pw/bands.py

```diff
@@ -259,7 +271,9 @@ def run_bands(self):
         inputs.metadata.call_link_label = 'bands'
         inputs.kpoints = self.ctx.bands_kpoints
         inputs.pw.structure = self.ctx.current_structure
-        inputs.pw.parent_folder = self.ctx.current_folder
+        # override the parent_folder if the scf was run
+        if 'scf' in self.inputs:
+            inputs.pw.parent_folder = self.ctx.current_folder
```

Comment:
How would the `parent_folder` be defined though? It is excluded from the `bands` input namespace, and if scf is skipped this will therefore not be set?

Reply:
Thanks for pointing out this. I didn't notice that the `parent_folder` is excluded. I run a simple example with Si by setting the `parent_folder`, it's strange that no error is reported, and the process finished successfully! Here is the process detail:
```console
$ verdi process show 28038
Property     Value
-----------  ------------------------------------
type         PwBandsWorkChain
state        Finished [0]
pk           28038
uuid         94b2d315-6a72-4f73-ae87-5a3b50d86d07
label
description
ctime        2024-03-21 15:34:17.077194+00:00
mtime        2024-03-21 15:34:24.276367+00:00

Inputs                  PK     Type
----------------------  -----  -------------
bands
    pw
        code            216    InstalledCode
        pseudos
            Si          19     UpfData
        parameters      28034  Dict
        parent_folder   28029  RemoteData
    max_iterations      28035  Int
bands_kpoints_distance  28036  Float
clean_workdir           28037  Bool
structure               28020  StructureData

Outputs                 PK  Type
-------------------  -----  -------------
band_parameters      28054  Dict
band_structure       28052  BandsData
primitive_structure  28042  StructureData
seekpath_parameters  28040  Dict

Called       PK  Type
--------  -----  ---------------------------
seekpath  28039  seekpath_structure_analysis
bands     28046  PwBaseWorkChain

Log messages
---------------------------------------------
There are 3 log messages for this calculation
Run 'verdi process report 28038' to see them
```
You see that the `parent_folder` is set. Here is the node graph
![Screenshot from 2024-03-21 16-50-37](https://github.com/aiidateam/aiida-quantumespresso/assets/11457659/46441ff1-c9ee-4f17-8eb8-e0733dc8654b)